images/egress/dns-proxy/egress_dns_proxy_test.go

```diff
@@ -0,0 +1,255 @@
+80 11.12.13.14
+
+# Port 8080 forwards to port 100 on example.com
+8080 example.com 100
+
+# Skip this rule for now
+# 9000 foo.com 200
+
+# End
+`,
+			frontends: []string{`
+frontend fe1
+    bind 1.2.3.4:80
+    default_backend be1`, `
+frontend fe2
+    bind 1.2.3.4:8080
+    default_backend be2`},
+			backends: []string{`
+backend be1
+    server dest1 11.12.13.14:80 check`, `
+backend be2
+    server dest2 example.com:100 check resolvers dns-resolver`},
+		},
+	}
+
+	for n, test := range tests {
+		cmd := exec.Command("./egress-dns-proxy.sh")
+		cmd.Env = []string{
+			fmt.Sprintf("EGRESS_DESTINATION=%s", test.dest),
+			fmt.Sprintf("EGRESS_SOURCE=%s", test.src),
+			fmt.Sprintf("EGRESS_DNS_PROXY_MODE=unit-test"),
+		}
+		outBytes, err := cmd.CombinedOutput()
+		if err != nil {
+			t.Fatalf("test %d unexpected error %v, output: %q", n+1, err, string(outBytes))
+		}
+		out := string(outBytes)
+		for _, frontend := range test.frontends {
+			if !strings.Contains(out, frontend) {
+				t.Fatalf("test %d expected frontend in output %q but got %q", n+1, frontend, out)
+			}
+		}
+		for _, backend := range test.backends {
+			if !strings.Contains(out, backend) {
+				t.Fatalf("test %d expected backend in output %q but got %q", n+1, backend, out)
+			}
+		}
+	}
+}
+
+func TestHAProxyFrontendBackendConfBad(t *testing.T) {
+	tests := []struct {
+		src  string
+		dest string
+		err  string
+	}{
+		{
+			src:  "",
+			dest: "80 11.12.13.14",
+			err:  "Must specify EGRESS_SOURCE",
+		},
+		{
+			src:  "not an IP addr",
+			dest: "80 11.12.13.14",
+			err:  "EGRESS_SOURCE must be IPv4 address",
+		},
+		{
+			src:  "1.2.3.4",
+			dest: "",
+			err:  "Must specify EGRESS_DESTINATION",
+		},
+		{
+			src:  "1.2.3.4",
+			dest: "80 11.12.13.14\ninvalid",
+			err:  "Bad destination 'invalid'",
+		},
+		{
+			src:  "1.2.3.4",
+			dest: "80 11.12.13.14\n8080 invalid",
+			err:  "Bad destination '8080 invalid'",
+		},
+		{
+			src:  "1.2.3.4",
+			dest: "99999 11.12.13.14",
+			err:  "Invalid port: 99999, must be in the range 1 to 65535",
+		},
+		{
+			src:  "1.2.3.4",
+			dest: "80 11.12.13.14 88888",
+			err:  "Invalid port: 88888, must be in the range 1 to 65535",
+		},
+		{
+			src:  "1.2.3.4",
+			dest: "80 11.12.13.14\n80 21.22.23.24 100",
+			err:  "Proxy port 80 already used, must be unique for each destination",
+		},
+	}
+
+	for n, test := range tests {
+		cmd := exec.Command("./egress-dns-proxy.sh")
+		cmd.Env = []string{
+			"EGRESS_DNS_PROXY_MODE=unit-test",
+			fmt.Sprintf("EGRESS_SOURCE=%s", test.src),
+			fmt.Sprintf("EGRESS_DESTINATION=%s", test.dest),
+		}
+		out, err := cmd.CombinedOutput()
+		out_lines := strings.Split(string(out), "\n")
+		got := out_lines[len(out_lines)-2]
+		if err == nil {
+			t.Fatalf("test %d expected error %q but got output %q", n+1, test.err, got)
+		}
+		if got != test.err {
+			t.Fatalf("test %d expected output %q but got %q", n+1, test.err, got)
+		}
+	}
+}
+
+func TestHAProxyDefaults(t *testing.T) {
```

Comment:
I was trying to catch the case where config file was overwritten by some write instead of append('>' instead of '>>' in the script). I will add the resolver test.